docs/design.md

```diff
@@ -0,0 +1,153 @@
+        networkMappings:
+        - source:
+            name: red
+            target: xyz
+            type: bridge
+status:
+ targetVmName: myvm # the name of the created virtual machine
+ conditions:
+   - lastHeartbeatTime: "2020-04-22T16:59:16Z"
+      lastTransitionTime: "2020-04-22T16:59:14Z"
+      message: Copying virtual machine disks
+      reason: CopyingDisks
+      status: "True"
+      type: Processing  # indicates if the VM import process is running
+ dataVolumes: # list of data volumes created for the VMs by the operator
+   - name: 8181ecc1-5db8-4193-9c92-3ddab3be7b05
+   - name: 8877699c-5c62-4b55-969b-45f8e07c25e9
+```
+
+The “source” element can be extended to represent additional source types for the VM import resource, such as VMWare and OVA, to which a tailored resource mapping will be required.
+
+### Progress monitoring
+
+Monitoring the progress of the vm import will be done by updating an annotation *vmimport.v2v.kubevirt.io/progress* with a numeric value in scale of 1 to 100. It is up to the operator to update the progress, in each step that is achieved, e.g:
+* progressStart        = "0"
+* progressCreatingVM   = "30"
+* progressCopyingDisks = "40"
+* progressStartVM      = "90"
+* progressDone         = "100"
+
+### Resource Mappings
+
+The mapping between ovirt to kubevirt resources is defined in ResourceMapping custom resource. The CR will contain sections for the mapping resources: network and storage. The example below demonstrates how multiple entities of each resource type can be declared and mapped.
+
+Each section specifies the mapping between ovirt entity to kubevirt’s entity, and allows to provide an additional piece of information, e.g. interface type if needed to complete the VM spec.
+
+The import VM operator will be responsible to deduce the configuration of the target VM based on the configuration of the source and perform the transformation in a way that will preserve the attributes of the source VM, e.g. boot sequence, MAC address, [run strategy](https://kubevirt.io/user-guide/docs/latest/creating-virtual-machines/run-strategies.html), [domain specification](https://kubevirt.io/api-reference/v0.26.1/definitions.html#_v1_domainspec) and hostname if available on ovirt by the guest agent.
+
+```yaml
+apiVersion: v2v.kubevirt.io/v1alpha1
+kind: ResourceMapping
+metadata:
+ name: example-ovirtresourcemappings
+ namespace: example-ns
+Spec:
+  ovirt:
+    networkMappings:
+    - source:
+        name: red # maps of ovirt logic network to network attachment definition
+      target: xyz
+      type: bridge
+    - source:
+        name: ovirtmgmt
+      Target:
+        name: pod
+      type: pod
+    storageMappings:
+    - source:
+        name: ovirt_storage_domain_1 # maps ovirt storage domains to storage class
+      target: storage_class_1
+    diskMappings: # a mapping of a specific disk to storage class
+    - source:
+        id: 8181ecc1-5db8-4193-9c92-3ddab3be7b12
+        target: local-storage
```

Comment:
This needs to be moved to the VM Import CR, since diskMappings is supported only there. See https://github.com/kubevirt/vm-import-operator/pull/135/files#r413204897

Reply:
Oh ok, will do!